src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+
+                // TODO: Use DocumentSpanEntry
+                result.ContextText = new ClassifiedTextElement(excerptResult.ClassifiedSpans.Select(cspan =>
+                    new ClassifiedTextRun(cspan.ClassificationType, sourceText.ToString(cspan.TextSpan))));
+
+                result.ContextHighlightSpan = AsSpan(excerptResult.MappedSpan);
+                // TODO: add definition context and its highlight
+            }
+
+            // Set location so that Editor can navigate to the result
+            var mappedDocumentSpan = await FindUsagesUtilities.TryMapAndGetFirstAsync(documentSpan, sourceText, cancellationToken).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
+            {
+                var location = mappedDocumentSpan.Value;
+
+                result.RelativePath = documentSpan.Document.FilePath;
+                result.LineNumber = location.LinePositionSpan.Start.Line;
+                result.CharacterNumber = location.LinePositionSpan.Start.Character;
+
+                result.PersistentSpan = context.SymbolSource.ServiceProvider.PersistentSpanFactory.Create(
+                    documentSpan.Document.FilePath,
+                    location.LinePositionSpan.Start.Line,
+                    location.LinePositionSpan.Start.Character,
+                    location.LinePositionSpan.End.Line,
+                    location.LinePositionSpan.End.Character,
+                    SpanTrackingMode.EdgeInclusive);
+            }
+
+            return result;
+        }
+
+        private static Span AsSpan(TextSpan sourceSpan)
+        {
+            return new Span(sourceSpan.Start, sourceSpan.Length);
+        }
+
+        public string RelativePath { get; private set; }
+
+        public int LineNumber { get; private set; }
+
+        public int CharacterNumber { get; private set; }
+
+        public IPersistentSpan PersistentSpan { get; private set; }
+
+        public ClassifiedTextElement ClassifiedDefinition { get; private set; }
+
+        public Span ContextHighlightSpan { get; private set; }
+
+        public string DocumentGuid { get; private set; }
+
+        public string ProjectGuid { get; private set; }
+
+        public ClassifiedTextElement ContextText { get; private set; }
+
+        public ImageElement Icon { get; private set; }
+
+        public string ProjectName { get; private set; }
+
+        public SymbolSearchResult Definition { get; private set; }
+
+        public bool IsDefinition { get; private set; }
+
+        // Private, because these fields are accessed from TryGetValue until LSP spec is finalized
+
+        private string Kind { get; set; }
+
+        private bool IsReadFrom { get; set; }
+
+        private bool IsWrittenTo { get; set; }
+
+        private string ContainingTypeName { get; set; }
+
+        private string ContainingMemberName { get; set; }
+
+        /// <summary>
+        /// Allows us to pass data whose API hasn't been fully fleshed out in the LSP standard
```

Comment:
Why does the LSP spec have anything to do with us consuming the editor API here? The comment seems off. I'm also not sure why we wouldn't already have a strongly typed interface for ContainingMember/TypeName/IsReadFrom/IsWrittenTo? I can understand wanting the backdoor for something additional down the road, but it seems odd that the API can't handle our current known features?